src/data/int/log.lean

```diff
@@ -0,0 +1,157 @@
+/-
+Copyright (c) 2022 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import data.nat.log
+import algebra.order.floor
+import algebra.field_power
+
+/-!
+# Integer logarithms in a field with respect to a natural base
+
+This file defines two `ℤ`-valued analogs of the logarithm of `n : R` with base `b : ℕ`:
+
+* `int.log b n`: Lower logarithm, or floor **log**. Greatest `k` such that `↑b^k ≤ n`.
+* `int.clog b n`: Upper logarithm, or **c**eil **log**. Least `k` such that `n ≤ ↑b^k`.
+
```

Comment:
Could you explain here the use of `int.log`/`int.clog` (namely, finding the significant digits in the decimal expansion)?

Reply:
Arguably, their use is already justified by the fact that they are the upper/lower adjoints (in terms of Galois connections) to exponentiation.

Reply:
I don't quite have the result that they're upper / lower adjoint though.

Any chance you could have a go @YaelDillies? The result I want is
```
/-- `zpow b` and `int.log b` (almost) form a Galois connection. -/
lemma zpow_le_iff_le_log {b : ℕ} (hb : 1 < b) {x : ℤ} {y : R} (hy : 0 < y) :
  (b : R) ^ x ≤ y ↔ x ≤ log b y :=
sorry
```
or similar

Reply:
I pushed a proof, but it's ugly.

Reply:
I pushed a proof, but it's ugly.

Reply:
I've added a digits example to the docstring too